### ISSUE TITLE
Run Postgraphile in Docker with Plugins

### ISSUE THREAD

**Issue by johnknelsonintific** [graphile/postgraphile#1126]
Hi All,

I'm submitting a:

* [ ] bug report
* [ ] feature request
* [X ] question 

PostGraphile version:

4.4.1

I am wanting to dockerize postgraphile to include in a docker-compose.yml file I am writing, but I also want to roll in a custom plugin I wrote. Is this possible? I have tried a few ways to do that so far, which I'll mention briefly:

1. Build a docker image based on Node JS and install all necessary packages, copy my_plugin.js over, and run postgraphile with command:

```
CMD postgraphile -c ${POSTGRES_CONNECTION_STRING} --subscriptions --append-plugins postgraphile-plugin-connection-filter --append-plugins `pwd`/my_plugin.js
```
In this case I can run the container (I am testing in a compose file that brings up postgresql as well, so that is taken care of) and see in the output that postgraphile is started up correctly, but I cannot access it at localhost:5000, even when port mapping (`-p 5000:5000`). I can, however, `exec` into the container and run `curl http://localhost:5000/graphiql` and get the expected graphiql web page html back, I just can't do it via the exposed port. Has anyone else tried installing and launching in their own Dockerfile?  I'm stuck on where the networking is going awry.

2. I am able to use the official `graphile/postgraphile` image directly in my compose file, which fixes my networking issue (I'm able to hit localhost:5000/graphiql and the endpoints fine from my host), but in this case, I don't know how to copy over and use my_plugin.js. I also don't know if I'd be able to use other third party plugins like `postgraphile-plugin-connection-filter`. It seems like that's not a current feature from my reading on [this Github issue](https://github.com/graphile/postgraphile/issues/1107). Am I missing the correct way to bundle plugins with the official Docker image?

3. I attempted to use the [official image ](https://hub.docker.com/r/graphile/postgraphile) as a base image `graphile/postgraphile:latest` for my own Dockerfile. This may expose my newness to Docker, but it looks like in that case it's missing libraries that would be copied over in the build... I suppose it doesn't actually build the whole file from source when I include it as `FROM graphile/postgraphile:latest` in my own Dockerfile? Can anyone confirm this is a bad practice?

Is what I want to achieve possible? If anyone can point out any errors I might have with anything I've tried I'd really appreciate it. And if there is another, more standard way, then that would be perfect! Thanks!

**Comment by johnknelsonintific** [comment]
**Solved**

The networking issue when kicking off postgraphile from within a Docker container was solved by adding `-n 0.0.0.0` to my launch command. The default is `localhost` which was not producing the desired behavior.

**Comment by benjie** [comment]
Thanks for sharing @johnknelsonintific; in our Docker build we solve this by adding this PostGraphileRC file:

https://github.com/graphile/postgraphile/blob/master/docker/.postgraphilerc.js
